### PR TITLE
Introducing Auth0 Hooks

### PR DESCRIPTION
**Pull request by tjanczuk**
New post to introduce Auth0 Hooks. Ideally this would go out this Friday, 2/17, which is when we plan to enable the feature.

### CODE REVIEW

**Comment by woloski** [comment]
Let's add a couple of screenshots. 

One of the dashboard

image

Another one of the editor

File: _posts/2017-02-15-introducing-auth0-hooks.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+* Integrated secret management improves the security of your code by providing a mechanism to securely store secrets while making them conveniently available in code.  
+* Integrated runner allows you to test your code without leaving the webtask editor.  
+* Real-time logs simplify debugging by streaming the output generated by your code.  
+* Using the [Auth0 CLI](https://auth0.com/docs/auth0-hooks/cli) you can scaffold, create, activate, and deactive hooks from the command line.  
+
+## What can you do today
+
+The initial release of Auth0 Hooks supports customizing the behavior of Auth0 at three new extensibility points:  
+
+* [Client Credentials Exchange](https://auth0.com/docs/auth0-hooks/extensibility-points/client-credentials-exchange) allows you to change the scopes and add custom claims to issued access tokens.
```
Comment (ggoodman) [inline _posts/2017-02-15-introducing-auth0-hooks.md]: What about password exchange?

File: _posts/2017-02-15-introducing-auth0-hooks.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+## Better developer experience
+
+While Auth0 Hooks are building on the same underlying [Webtask](https://webtask.io) technology we have developed to run Auth0 Rules, several aspects of the developer experience are improved: 
+
+* Using the [management dashboard](https://auth0.com/docs/auth0-hooks/dashboard) you can create, move in and out of production, and edit hooks for selected extensibility points in the Auth0 platform. 
+
+![Auth0 Hooks dashboard](https://cdn.auth0.com/hooks/assets/manage-hooks.png)  
+
+* You edit hook code in the Webtask Editor which offers a much richer featureset compared to the experiene you are used to with Auth0 Rules.
```
Comment (glennblock) [inline _posts/2017-02-15-introducing-auth0-hooks.md]: You edit hooks in the Webtask Editor, which offers a much richer featureset compared to the experience you are used to with Auth0 rules

experience

File: _posts/2017-02-15-introducing-auth0-hooks.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+![Auth0 Hooks dashboard](https://cdn.auth0.com/hooks/assets/manage-hooks.png)  
+
+* You edit hook code in the Webtask Editor which offers a much richer featureset compared to the experiene you are used to with Auth0 Rules.  
+* Syntax completion allows you write the code faster without referring to documentation.  
+* Integrated secret management improves the security of your code by providing a mechanism to securely store secrets while making them conveniently available in code.  
+* Integrated runner allows you to test your code without leaving the webtask editor.  
+* Real-time logs simplify debugging by streaming the output generated by your code.  
+* Using the [Auth0 CLI](https://auth0.com/docs/auth0-hooks/cli) you can scaffold, create, activate, and deactive hooks from the command line.  
+
```
Comment (glennblock) [inline _posts/2017-02-15-introducing-auth0-hooks.md]: Github integration?

File: _posts/2017-02-15-introducing-auth0-hooks.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+## What can you do today
+
+The initial release of Auth0 Hooks supports customizing the behavior of Auth0 at three new extensibility points:  
+
+* [Client Credentials Exchange](https://auth0.com/docs/auth0-hooks/extensibility-points/client-credentials-exchange) allows you to change the scopes and add custom claims to issued access tokens.  
+* [Pre User Registration](https://auth0.com/docs/auth0-hooks/extensibility-points/pre-user-registration) allows you to intercept creation of a new dabatase user to enforce custom password policy, or employ application specific logic to prevent the signup.  
+* [Post User Registration](https://auth0.com/docs/auth0-hooks/extensibility-points/post-user-registration) allows you to perform any actions as a result of a succcessful creation of a new database user, e.g. send a message to Slack, or create a record in your CRM system.  
+
+We are going to be adding many more extensibility points in the Auth0 platform using the Auth0 Hooks mechanism in the future.
```
Comment (glennblock) [inline _posts/2017-02-15-introducing-auth0-hooks.md]: This is just the beginning. We are going to be adding many more extensibility.......!

File: _posts/2017-02-15-introducing-auth0-hooks.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+* [Pre User Registration](https://auth0.com/docs/auth0-hooks/extensibility-points/pre-user-registration) allows you to intercept creation of a new dabatase user to enforce custom password policy, or employ application specific logic to prevent the signup.  
+* [Post User Registration](https://auth0.com/docs/auth0-hooks/extensibility-points/post-user-registration) allows you to perform any actions as a result of a succcessful creation of a new database user, e.g. send a message to Slack, or create a record in your CRM system.  
+
+We are going to be adding many more extensibility points in the Auth0 platform using the Auth0 Hooks mechanism in the future.   
+
+## Auth0 Hooks vs Auth0 Rules
+
+Introduction of Auth0 Hooks does not affect any existing Auth0 Rules. Your rules continue to work unchanged. 
+
+Auth0 Hooks provide a foundation for a new extensibility mechanism in Auth0. All future extensibility points in the platfrom will build on top of Auth0 Hooks. We are also working on adding support in Auth0 Hooks for the same set of customization scenarios that are addressed by Auth0 Rules today.
```
Comment (glennblock) [inline _posts/2017-02-15-introducing-auth0-hooks.md]: We are also planning to add support in Auth0 Hooks for the same things you use Auth0 Rules for today.

File: _posts/2017-02-15-introducing-auth0-hooks.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+## Auth0 Hooks vs Auth0 Rules
+
+Introduction of Auth0 Hooks does not affect any existing Auth0 Rules. Your rules continue to work unchanged. 
+
+Auth0 Hooks provide a foundation for a new extensibility mechanism in Auth0. All future extensibility points in the platfrom will build on top of Auth0 Hooks. We are also working on adding support in Auth0 Hooks for the same set of customization scenarios that are addressed by Auth0 Rules today. 
+
+## Differences with Auth0 Rules
+
+If you have been using Auth0 Rules before, these are some of the key differences in the development experience when moving on to Auth0 Hooks:
```
Comment (glennblock) [inline _posts/2017-02-15-introducing-auth0-hooks.md]: How about just putting a table here with 3 columns, Item, Auth0 Rules and Auth0 Hooks? Better than just reading a narrative for quickly scanning differences (imo)

**Pull request merged**